ksqldb-engine/src/main/java/io/confluent/ksql/engine/rewrite/AstSanitizer.java

```diff
@@ -192,5 +206,22 @@ public static Statement sanitize(final Statement node, final MetaStore metaStore
           )
       );
     }
+
+    @Override
+    public Optional<Expression> visitLambdaExpression(
+        final LambdaFunctionExpression expression,
+        final Object ctx
+    ) {
+      dataSourceExtractor.getAllSources().forEach(aliasedDataSource -> {
+        for (String argument : expression.getArguments()) {
+          if (aliasedDataSource.getDataSource().getSchema().columns().stream()
+              .map(column -> column.name().text()).collect(Collectors.toList())
+              .contains(argument)) {
+            throw new KsqlException("Lambda argument can't be a column name.");
```

Comment:
For my own education: could you elaborate the rationale of this verification? Maybe give me an example KSQL statement with lambda func?